paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -374,17 +374,24 @@ def mesos_cpu_metrics_provider(
     monkey.patch_socket()
     jobs = [gevent.spawn(task.stats_callable) for task in mesos_tasks]
     gevent.joinall(jobs, timeout=60)
-    mesos_tasks = dict(zip([task['id'] for task in mesos_tasks], [job.value for job in jobs]))
+    mesos_tasks_first_run = dict(zip([task['id'] for task in mesos_tasks], [job.value for job in jobs]))
 
     current_time = int(datetime.now().strftime('%s'))
     time_delta = current_time - last_time
 
+    # Mesos slave statistics endpoint returns a crazy CPU value when a container is
+    # in the process of being killed. So we fetch twice, and use the lower of the two.
```

Comment:
Is it possible for us to make the second request too quickly and get the crazy CPU value again? Do we need any sort of configurable delay in the middle?